aviary/cgcnn/data.py

```diff
@@ -24,140 +25,101 @@ def __init__(
             dmin (float, optional): minimum distance in Gaussian basis. Defaults to 0.
             step (float, optional): increment size of Gaussian basis. Defaults to 0.2.
         """
-        if len(identifiers) != 2:
-            raise AssertionError("Two identifiers are required")
-        if len(inputs) != 2:
-            raise AssertionError("One input column required are required")
-
-        self.inputs = list(inputs)
         self.task_dict = task_dict
         self.identifiers = list(identifiers)
 
         self.radius = radius
         self.max_num_nbr = max_num_nbr
 
-        if elem_emb in ["matscholar200", "cgcnn92", "megnet16", "onehot112"]:
-            elem_emb = f"{PKG_DIR}/embeddings/element/{elem_emb}.json"
-        elif not os.path.exists(elem_emb):
-            raise AssertionError(f"{elem_emb} does not exist!")
+        if elem_embedding in ("matscholar200", "cgcnn92", "megnet16", "onehot112"):
+            elem_embedding = f"{PKG_DIR}/embeddings/element/{elem_embedding}.json"
+        elif not os.path.exists(elem_embedding):
+            raise ValueError(f"{elem_embedding} does not exist!")
 
-        with open(elem_emb) as f:
+        with open(elem_embedding) as f:
             self.elem_features = json.load(f)
 
         for key, value in self.elem_features.items():
             self.elem_features[key] = np.array(value, dtype=float)
+            if not hasattr(self, "elem_emb_len"):
+                self.elem_emb_len = len(value)
+            elif self.elem_emb_len != len(value):
+                raise ValueError("Element embedding length mismatch!")
 
-        self.elem_emb_len = len(list(self.elem_features.values())[0])
-
-        self.gdf = GaussianDistance(dmin=dmin, dmax=self.radius, step=step)
+        self.gdf = GaussianDistance(dmin=dmin, dmax=radius, step=step)
         self.nbr_fea_dim = self.gdf.embedding_size
 
         self.df = df
-        self.df["Structure_obj"] = self.df[self.inputs].apply(get_structure, axis=1)
+        self.structure_col = structure_col
+
+        all_isolated = []
+        some_isolated = []
+
+        for material_id, struct in tqdm(
+            zip(self.df[identifiers[0]], self.df[structure_col]),
+            total=len(df),
+            desc="Pre-check that all structures are valid, i.e. none have isolated atoms.",
+            disable=None,
+        ):
+            self_idx, nbr_idx, _ = get_structure_neighbor_info(
+                struct, radius, max_num_nbr
+            )
+
+            if 0 in (len(self_idx), len(nbr_idx)):
+                all_isolated.append(material_id)
+            if set(self_idx) != set(range(len(struct))):
+                some_isolated.append(material_id)
+
+        isolated = set(all_isolated + some_isolated)
+        if len(isolated) > 0:
+            # drop the data points that do not give rise to dense crystal graphs
+            # TODO next line requires identifiers[0] == df.index, bad assumption
```

Comment:
this no longer applies?

Reply:
it _does_ still apply. i forgot about this. will think of a better solution